src/verify_cert.rs

```diff
@@ -204,14 +204,34 @@ fn check_signatures(
 }
 
 fn check_crl(
-    _supported_sig_algs: &[&SignatureAlgorithm],
+    supported_sig_algs: &[&SignatureAlgorithm],
     cert: &Cert,
-    _issuer_subject: untrusted::Input,
-    _issuer_spki: untrusted::Input,
+    issuer_subject: untrusted::Input,
+    issuer_spki: untrusted::Input,
     crl_provider: &dyn CrlProvider,
 ) -> Result<(), Error> {
-    let _relevant_crl = crl_provider.crl_for_cert(cert);
-    todo!();
+    assert_eq!(cert.issuer, issuer_subject);
+
+    let relevant_crl = crl_provider.crl_for_cert(cert);
+    match relevant_crl {
+        None => Ok(()),
+        Some(crl) => {
+            // Verify the CRL signature with the issuer SPKI.
+            // TODO(XXX): consider whether we can refactor so this happens once up-front, instead
+            //            of per-lookup.
+            //            https://github.com/rustls/webpki/issues/81
+            signed_data::verify_signed_data(supported_sig_algs, issuer_spki, &crl.signed_data)?;
+
+            // TODO(@cpu): Verify the issuer has no KU, or KU with cRLSign.
+
+            // Try to find the cert serial in the verified CRL contents.
+            let cert_serial = cert.serial.as_slice_less_safe();
+            return match crl.find_serial(cert_serial) {
```

Comment:
Style nit: no need for a `return` keyword here.

Reply:
Fixed.